src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/AwsSdk/DynamoDbRequestHandler.cs

```diff
@@ -112,5 +35,16 @@ public static AfterWrappedMethodDelegate HandleDynamoDbRequest(InstrumentedMetho
                 :
                 Delegates.GetDelegateFor(segment);
         }
+
+        private static string GetOperationNameFromRequestType(string requestType)
+        {
+            if (_operationNameCache.ContainsKey(requestType))
+            {
+                return _operationNameCache[requestType];
+            }
+            var operationName = requestType.Replace("Request", string.Empty).ToSnakeCase();
+            _operationNameCache.Add(requestType, operationName);
```

Comment:
Do we need to be worried about concurrency with this cache? It currently uses a regular dictionary which is not thread safe.